packages/metals-vscode/src/extension.ts

```diff
@@ -228,9 +229,24 @@ async function fetchAndLaunchMetals(
     javaVersion,
     metalsDirPath,
     context.extensionPath,
-    outputChannel
+    outputChannel,
+    forceCoursierJar
   );
 
+  const canRetryWithJar =
+    javaHome && !coursier.endsWith(".jar") && !forceCoursierJar;
+
+  function retry(error: any): Promise<any> {
+    if (canRetryWithJar) {
+      outputChannel.appendLine(
+        "Trying again with the embedded coursier. This might take longer."
+      );
+      return fetchAndLaunchMetals(context, serverVersion, javaVersion, true);
```

Comment:
If you rerun the whole `fetchAndLaunchMetals` it will again try to download coursier, this seems redundant.